src/main/java/net/backupcup/mcde/screen/util/ScreenWithSlots.java

```diff
@@ -2,11 +2,11 @@
 
 import java.util.Optional;
 
-import net.backupcup.mcde.util.Slots;
+import net.backupcup.mcde.util.SlotPosition;
 import net.minecraft.client.util.math.MatrixStack;
 
 public interface ScreenWithSlots {
-    Optional<Slots> getOpened();
-    void setOpened(Optional<Slots> opened);
+    Optional<SlotPosition> getOpened();
+    void setOpened(Optional<SlotPosition> opened);
```

Comment:
Description is key with `Optionals` since they can be so many things. Since it is a type of `opened`, consider `setOpenedSlotPosition()`. There is nothing wrong with descriptive names, especially if you have access to intellisense.